src/server/qgsserversettings.h

```diff
@@ -66,7 +66,8 @@ class SERVER_EXPORT QgsServerSettingsEnv : public QObject
       QGIS_SERVER_WMS_MAX_HEIGHT, //! Maximum height for a WMS request. The most conservative between this and the project one is used (since QGIS 3.8)
       QGIS_SERVER_WMS_MAX_WIDTH, //! Maximum width for a WMS request. The most conservative between this and the project one is used (since QGIS 3.8)
       QGIS_SERVER_API_RESOURCES_DIRECTORY, //! Base directory where HTML templates and static assets (e.g. images, js and css files) are searched for (since QGIS 3.10).
-      QGIS_SERVER_API_WFS3_MAX_LIMIT //! Maximum value for "limit" in a features request, defaults to 10000 (since QGIS 3.10).
+      QGIS_SERVER_API_WFS3_MAX_LIMIT, //! Maximum value for "limit" in a features request, defaults to 10000 (since QGIS 3.10).
+      QGIS_REVEAL_SERVER_SETTINGS //! Enables QGIS Server legacy request GetServerSettings
```

Comment:
Can you please rename it to something that starts with QGIS_SERVER_ like QGIS_SERVER_ENABLE_GETSERVERSETTINGS?

Reply:
What about `QGIS_SERVER_DEBUG_MODE` (assuming that it makes it clear that this is not for production)?

Reply:
> What about `QGIS_SERVER_DEBUG_MODE` (assuming that it makes it clear that this is not for production)?

@m-kuhn Both option are good (thanks!), but maybe `QGIS_SERVER_ENABLE_GETSERVERSETTINGS` is less ambiguous. People might think that `QGIS_SERVER_DEBUG_MODE` has other side effects, like running in DEBUG MODE, instead of production mode. We already have `QGIS_DEBUG` and this `QGIS_SERVER_DEBUG_MODE` could be understood as the same, but for running `qgis_mapserv.fcgi`.

Reply:
Good point. Let's just document this clearly with comments in all the examples that demonstrate setting up a server.